### PR TITLE
#371 better encoding support

### PR DESCRIPTION
**Pull request by metsjeesus**
I just explain problem first.
I got situation where irc server sends both utf8 and iso-8859-1 messages. And i want both to be human readable. Problem is, if i use utf8, i cant see or recover iso-8859-1 messages. But if i use iso-8859-1 as encoding, i cant see utf8 messages but i can recover them.  Using iso-8859-1 makes other problem, if outgoing messages are coded on iso-8859-1, you cant send utf8 text correctly like `Iñtërnâtiônàlizætiøn☃💩'`

So, to fix it, you need different encodings to irc incoming messages and outgoing messages. And top of that, you need to get rid Mojibake, miscoded characters.  My solution is backward compatible encoding string in irc-framework configuration. So you can configure it in lounge for every network you connect.

`utf8`- normal way
`iso-8859-1/utf8`- incoming messages are coded in iso-8859-1, outgoing in utf8
`iso-8859-1:auto`- incoming and outgoing is iso-8859-1 and gets rid of mojibikes on incoming messages
`iso-8859-1:auto/utf8`- incoming messags are coded iso-8859-1 and gets rid of mojibikes, outgoing messages are coded utf8

autocorrect reqursively makes text human readable. That means certain text combinations, like `Ã¤` automatcly will be converted to `ä` and text will lose that information. For humans its good, but not for computers. 

;tldr made thing that fixes utf8 - iso-8859-1 Mojibake characters in node.

### CODE REVIEW

File: defaults/config.js [hunk 1]
```diff
@@ -194,7 +194,15 @@ module.exports = {
 		// @default  6697
 		//
 		port: 6697,
-
+		
+		//
+		// Encoding
+		//
+		// @type     string
```
Comment (MaxLeiter) [inline defaults/config.js]: Could you list the examples of possible encodings here? And the proper formatting for them?

File: package.json [hunk 2]
```diff
@@ -50,7 +50,7 @@
     "event-stream": "3.3.2",
     "express": "4.13.4",
     "fs-extra": "0.30.0",
-    "irc-framework": "2.5.0",
+    "irc-framework": "git://github.com/metsjeesus/irc-framework.git#encoding",
```
Comment (MaxLeiter) [inline package.json]: I think it's worth waiting for the merge upstream, rather than switching to a fork

Comment (metsjeesus) [inline package.json]: yes, its worth waiting, but its only way to test it though.

File: defaults/config.js (hunk 1 above)
Comment (metsjeesus) [inline defaults/config.js]: done and done

**Comment by xPaw** [comment]
I will close this PR for now as it won't work without irc-fw changes, and seeing the discussion doesn't make it seem like it will happen any time soon.

**Pull request closed without merging**